OsmAnd/src/net/osmand/plus/configmap/tracks/viewholders/TrackViewHolder.java

```diff
@@ -252,20 +253,27 @@ private void appendDistanceDescription(@NonNull SpannableStringBuilder builder,
 		appendFolderName(builder, trackItem, shouldShowFolder);
 	}
 
+	private float getDistance(@NonNull TrackItem trackItem, @NonNull GpxTrackAnalysis analysis) {
+		GpxDataItem dataItem = trackItem.getDataItem();
+		boolean joinSegments = dataItem != null && (boolean) dataItem.getParameter(GpxParameter.JOIN_SEGMENTS);
+		return joinSegments ? analysis.getTotalDistance() : analysis.getTotalDistanceWithoutGaps();
+	}
```

Comment:
We may need the same info in other places, we could add a universal method to GpxTrackAnalysis or TrackItem instead of UI